### PR TITLE
Fixes #1617: Logging for apoc.periodic.*

### LINKED ISSUE #1617: Logging for apoc.periodic.*
## Feature description (Mandatory)
It would be great to have more detail on execution of the apoc.periodic.* - specifically periodic.iterate and probably parallel apocs too, could we generate additional logging to show progress of batches e.g.

Processing [batch x] - [size y]?

Not sure whether we'd benefit start/end messages for each batch (to give information on length of time a batch takes? or just enough to know there is progress

## Considered alternatives
Currently we can add apoc.log.* to the periodic iterate command to get some logging

### PR DESCRIPTION
**Pull request by vga91**
Fixes #1617

One sentence summary of the change.

## Proposed Changes (Mandatory)

A brief list of proposed changes in order to fix the issue:

  - Added unique identifier to periodic procedures
  - Added logs when in `neo4j.conf` we have `dbms.logs.debug.level=DEBUG`

### CODE REVIEW

File: core/src/main/java/apoc/periodic/Periodic.java [hunk 1]
```diff
@@ -64,6 +64,10 @@
         AtomicInteger failedBatches = new AtomicInteger();
         Map<String,Long> batchErrors = new ConcurrentHashMap<>();
 
+        String periodicId = UUID.randomUUID().toString();
+        if (log.isDebugEnabled()) {
+            log.debug("Starting periodic commit from `%s` in separate thread with id: `%s`", statement, periodicId);
```
Comment (fbiville) [inline core/src/main/java/apoc/periodic/Periodic.java:69]: nit: might be worth removing the `if` and use the `{}` placeholder syntax of slf4j if applicable

**Review by fbiville — APPROVED** [review]

File: core/src/main/java/apoc/periodic/Periodic.java (hunk 1 above)
Comment (vga91) [reply core/src/main/java/apoc/periodic/Periodic.java]: Removed `if` statements, with `{}` it doesn't seem to work with `neo4j.logging.Log`..

Comment (fbiville) [reply core/src/main/java/apoc/periodic/Periodic.java]: If no access to that, might be better to keep those `if`s for consistency? I'm not sure it really affects performance too much, as it just created a few String instances here and there, but I haven't monitored that all

Comment (vga91) [reply core/src/main/java/apoc/periodic/Periodic.java:68]: You're right, better keep them. Restored

**Pull request merged**